test/parallel/test-promises-unhandled-rejections.js

```diff
@@ -0,0 +1,591 @@
+    var a = Promise.reject(e);
+    process.nextTick(function() {
+      Promise.resolve().then(function() {
+        process.nextTick(function() {
+          Promise.resolve().then(function() {
+            a.catch(function() {});
+          });
+        });
+      });
+    });
+  }, 0);
+});
+
+asyncTest(function(done) {
+  var e = new Error();
+  onUnhandledSucceed(done, function(reason, promise) {
+    assert.strictEqual(e, reason);
+    assert.strictEqual(p, promise);
+  });
+  var p = Promise.reject(e);
+  setImmediate(function() {
+    Promise.resolve().then(function () {
+      p.catch(function(){});
+    });
+  });
+});
+
+asyncTest(function(done) {
+  onUnhandledSucceed(done, function(reason, promise) {
+    assert.strictEqual(undefined, reason);
+    assert.strictEqual(p, promise);
+  });
+  setImmediate(function() {
+    Promise.resolve().then(function () {
+      Promise.resolve().then(function () {
+        Promise.resolve().then(function () {
+          Promise.resolve().then(function () {
+            p.catch(function(){});
+          });
+        });
+      });
+    });
+  });
+  var p = Promise.reject();
+});
+
+asyncTest(function(done) {
+  onUnhandledSucceed(done, function(reason, promise) {
+    assert.strictEqual(undefined, reason);
+    assert.strictEqual(p, promise);
+  });
+  var p = Promise.reject();
+  setImmediate(function() {
+    Promise.resolve().then(function () {
+      Promise.resolve().then(function () {
+        Promise.resolve().then(function () {
+          Promise.resolve().then(function () {
+            p.catch(function(){});
+          });
+        });
+      });
+    });
+  });
+});
+
+asyncTest(function(done) {
+  var e = new Error();
+  onUnhandledFail(done);
+  var a = Promise.reject(e);
+  process.nextTick(function() {
+    Promise.resolve().then(function() {
+      process.nextTick(function() {
+        Promise.resolve().then(function() {
+          a.catch(function() {});
+        })
+      })
+    })
+  });
+});
+
+asyncTest(function(done) {
+  var e = new Error();
+  onUnhandledFail(done);
+  setImmediate(function() {
+    var a = Promise.reject(e);
+    process.nextTick(function() {
+      Promise.resolve().then(function() {
+        process.nextTick(function() {
+          Promise.resolve().then(function() {
+            a.catch(function() {});
+          })
+        })
+      })
+    });
+  });
+});
+
+asyncTest(function(done) {
+  var d = domain.create();
+  var domainReceivedError;
+  d.on('error', function(e) {
+    domainReceivedError = e;
+  });
+  d.run(function() {
+    var e = new Error('error');
+    var domainError = new Error('domain error');
+    onUnhandledSucceed(done, function(reason, promise) {
+      assert.strictEqual(reason, e);
+      assert.strictEqual(domainReceivedError, domainError);
+      d.dispose();
+    });
+    var a = Promise.reject(e);
+    process.nextTick(function() {
+      throw domainError;
+    });
+  });
+});
+
+asyncTest(function(done) {
```

Comment:
The tests whether nextTick is immediate scheduled while inside the event handler, if it wasn't, setTimeout would be called before nextTick. The test used to test difference between dates but @vkurchatkin  said this is ok too.